src/main/java/eu/arrowhead/client/library/ArrowheadService.java

```diff
@@ -453,6 +468,79 @@ public OrchestrationResponseDTO queryOrchestrationStore(final long systemId) {
 		return response.getBody();
 	}
 	
+	//-------------------------------------------------------------------------------------------------
+	/**
+	 * Make WSS connection with the specified service reachability details.
+	 *
+	 * @param handler WebSocket handler to use for message exchanges
+	 * @param address String value which represents the host where the service is available.
+	 * @param port int value which represents the port where the service is available
+	 * @param serviceUri String value which represents the URI where the service is available.
+	 * @param token (nullable) String value which represents the token for being authorized at the provider side if necessary. Token could be received in orchestration response per interface type.
+	 * @param queryParams (nullable) String... variable arguments which represent the additional key-value http(s) query parameters if any necessary. E.g.: "k1", "v1", "k2", "v2".
+	 * @return A WebSocketConnection manager that is used for all full-duplex communication
+	 *
+	 * @throws InvalidParameterException when service URL can't be assembled.
+	 * @throws CertificateException when a certificate or SSL error occurs
+	 * @throws ArrowheadException when the communication is managed via Gateway Core System and internal server error happened.
```

Comment:
I think at the moment ArrowheadExcpetion is thrown only when Keystore or TrustStroe cannot be opened, beacuse no connection attempt or data exchange is hapenning within this method. (Because WebSocketConnectionManager.start() is not called)

Reply:
Fixed